atomic_reactor/plugins/post_flatpak_create_oci.py

```diff
@@ -0,0 +1,308 @@
+        outfile = os.path.join(self.workflow.source.workdir, 'filesystem.tar.gz')
+
+        export_stream = self.tasker.d.export(container_id)
+        out_fileobj = open(outfile, "w")
+        compress_process = subprocess.Popen(['gzip', '-c'],
+                                            stdin=subprocess.PIPE,
+                                            stdout=out_fileobj)
+        in_tf = tarfile.open(fileobj=export_stream, mode='r|')
+        out_tf = tarfile.open(fileobj=compress_process.stdin, mode='w|')
+
+        for member in in_tf:
+            target_name = self._get_target_path(member.name)
+            if target_name is None:
+                continue
+
+            # flatpak tries to create files/.ref and behaves badly if the root directory
+            # isn't writable. It also has trouble upgrading if any directory is not
+            # user-writable.
+            member.mode |= 0200
+
+            member.name = target_name
+            if member.islnk():
+                # Hard links have full paths within the archive (no leading /)
+                link_target = self._get_target_path(member.linkname)
+                if link_target is None:
+                    self.log.debug("Skipping %s, hard link to %s", target_name, link_target)
+                    continue
+                member.linkname = link_target
+                out_tf.addfile(member)
+            elif member.issym():
+                # Symlinks have the literal link target, which will be
+                # relative to the chroot and doesn't need rewriting
+                out_tf.addfile(member)
+            else:
+                f = in_tf.extractfile(member)
+                out_tf.addfile(member, fileobj=f)
+
+        in_tf.close()
+        out_tf.close()
+        export_stream.close()
+        compress_process.stdin.close()
+        if compress_process.wait() != 0:
+            raise RuntimeError("gzip failed")
+        out_fileobj.close()
+
+        return outfile
+
+    def _export_filesystem(self):
+        image = self.workflow.image
+        self.log.info("Creating temporary docker container")
+        container_dict = self.tasker.d.create_container(image)
+        container_id = container_dict['Id']
+
+        try:
+            return self._export_container(container_id)
+        finally:
+            self.log.info("Cleaning up docker container")
+#            self.tasker.d.remove_container(container_id)
+
+    def _create_runtime_oci(self, tarfile, outfile):
+        info = self.source.flatpak_json
+
+        builddir = os.path.join(self.workflow.source.workdir, "build")
+        os.mkdir(builddir)
+
+        repo = os.path.join(self.workflow.source.workdir, "repo")
+        subprocess.check_call(['ostree', 'init', '--mode=archive-z2', '--repo', repo])
+
+        runtime_id = info['runtime']
+        runtime_version = info['runtime-version']
+
+        with open(os.path.join(builddir, 'metadata'), 'w') as f:
+            f.write("""[Runtime]
```

Comment:
`from textwrap import dedent` and `dedent("""long text""")` should be used here to preserve the identation